cmd/web-handlers.go

```diff
@@ -351,27 +351,65 @@ func (web *webAPIHandlers) SetAuth(r *http.Request, args *SetAuthArgs, reply *Se
 	if !isValidSecretKey.MatchString(args.SecretKey) {
 		return &json2.Error{Message: "Invalid Secret Key"}
 	}
+
 	cred := credential{args.AccessKey, args.SecretKey}
+
+	// Helper function - in case of an error in addition to a peer
+	// error, we return both together.
+	appendErr := func(peerErr, err *json2.Error) *json2.Error {
+		if peerErr == nil {
+			return err
+		}
+		return &json2.Error{
+			Message: "One or more errors happened!",
+			Data:    []*json2.Error{peerErr, err},
+		}
+	}
+
+	// Notify all Minio Peers to update credentials
+	errsMap := updateCredsOnPeers(cred)
+	var peerErr *json2.Error
+	if len(errsMap) != 0 {
+		peerErr = &json2.Error{
+			Message: "Could not update credentials on one or more peers.",
+			Data:    errsMap,
+		}
+	}
+
+	// Even if we fail to update some clients, update locally and
+	// let the user know
 	serverConfig.SetCredential(cred)
 	if err := serverConfig.Save(); err != nil {
-		return &json2.Error{Message: err.Error()}
+		return appendErr(
+			peerErr,
+			&json2.Error{Message: err.Error()},
+		)
 	}
 
 	jwt, err := newJWT(defaultJWTExpiry) // JWT Expiry set to 24Hrs.
 	if err != nil {
-		return &json2.Error{Message: err.Error()}
+		return appendErr(
+			peerErr,
+			&json2.Error{Message: err.Error()},
+		)
 	}
 
 	if err = jwt.Authenticate(args.AccessKey, args.SecretKey); err != nil {
-		return &json2.Error{Message: err.Error()}
+		return appendErr(
+			peerErr,
+			&json2.Error{Message: err.Error()},
+		)
 	}
 	token, err := jwt.GenerateToken(args.AccessKey)
 	if err != nil {
-		return &json2.Error{Message: err.Error()}
+		return appendErr(
+			peerErr,
+			&json2.Error{Message: err.Error()},
+		)
 	}
 	reply.Token = token
 	reply.UIVersion = miniobrowser.UIVersion
-	return nil
+	return peerErr
```

Comment:
``net/rpc`` expects rpc handlers to return ``nil`` for error if the call is successful. For readers to grasp that this code path represents success, return ``nil`` explicitly instead of relying on ``peerErr`` to hold ``nil`` value.